### ISSUE TITLE
use of unstable library feature 'assoc_char_funcs': recently added

### ISSUE THREAD

**Issue by Teomayo** [katharostech/bevy_retrograde#46]
The following error was found when trying to get setup on a Mac system on nightly rust. I noticed that currently there isn't support for Mac after the fact but the description outlined in the [Mac OS support ticket](https://github.com/katharostech/bevy_retrograde/issues/28) is different then what I ran into trying to get started. 

```
error[E0658]: use of unstable library feature 'assoc_char_funcs': recently added
   --> /Users/teodorcrnobrnja/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_retrograde_text-0.1.0/src/bdf.rs:140:33
    |
140 |                 let codepoint = char::from_u32(encoding).ok_or("Could not parse character encoding")?;
    |                                 ^^^^^^^^^^^^^^
    |
    = note: see issue #71763 <https://github.com/rust-lang/rust/issues/71763> for more information
    = help: add `#![feature(assoc_char_funcs)]` to the crate attributes to enable
```

After running into this I added `#![feature(assoc_char_funcs)]` to my `main.rs` file but there was no change. I am currently running the following toolchain: `info: default toolchain set to 'nightly-x86_64-apple-darwin'`

Another thing to note is that there is currently a discrepancy in the Sample on [crates.io](https://crates.io/crates/bevy_retrograde) and on Github, the Github version resulted in some errors regarding the `Transform` function which wasn't present on crates.io, so it should be noted the errors above happened when running the Sample on crates.io.

**Comment by zicklag** [comment]
It looks like that feature was stabilized in Rust 1.52.0. Running a `rustup update nightly` to update your rust version should fix that. This nightly Rust version worked for me but the latest stable should also work: `rustc 1.55.0-nightly (a85f584ae 2021-06-16)`.

> Another thing to note is that there is currently a discrepancy in the Sample on crates.io and on Github, the Github version resulted in some errors regarding the Transform function which wasn't present on crates.io, so it should be noted the errors above happened when running the Sample on crates.io.

Yes, we've got some big changes that we've made on master that haven't been published yet, but it should be compiling. If you want to use the master version and that doesn't compile after a Rust update, let me know and I'll look into it.
